src/main/kotlin/org/radarbase/push/integration/common/auth/DelegatedAuthValidator.kt

```diff
@@ -0,0 +1,33 @@
+package org.radarbase.push.integration.common.auth
+
+import org.glassfish.hk2.api.IterableProvider
+import org.radarbase.jersey.auth.Auth
+import org.radarbase.jersey.auth.AuthValidator
+import javax.ws.rs.container.ContainerRequestContext
+import javax.ws.rs.core.Context
+import javax.ws.rs.core.UriInfo
+
+class DelegatedAuthValidator(
+    @Context private val uriInfo: UriInfo,
+    @Context private val namedValidators: IterableProvider<AuthValidator>
+) : AuthValidator {
+
+    fun delegate(): AuthValidator {
+        return when {
+            uriInfo.matches(GARMIN_QUALIFIER) -> namedValidators.named(GARMIN_QUALIFIER).get()
+            // Add support for more as integrations are added
+            else -> throw IllegalStateException()
+        }
+    }
+
+    private fun UriInfo.matches(name: String): Boolean = this.path.contains(name)
```

Comment:
This should be extended to at least match `/$name/`, and preferably `"^/$basePath/integrations/$name/".toRegex()`, otherwise authentication injection may be done.